src/modules/cpu/kernel/warp_perspective.hpp

```diff
@@ -0,0 +1,2286 @@
+                    compute_warp_perspective_src_loc_next_term_avx(plocW, plocY, plocX, pSrcY, pSrcX, pPerspectiveMatrixTerm6Incr, pPerspectiveMatrixTerm3Incr, pPerspectiveMatrixTerm0Incr, roiHalfHeightVec, roiHalfWidthVec);
+                    dstPtrTemp += vectorIncrementPkd;
+                }
+                locW += (perspectiveMatrix_f9->data[6] * vectorLoopCount);
+                locY += (perspectiveMatrix_f9->data[3] * vectorLoopCount);
+                locX += (perspectiveMatrix_f9->data[0] * vectorLoopCount);
+                srcX = (locX/locW) + roiHalfWidth;
+                srcY = (locY/locW) + roiHalfHeight;
+                for (; vectorLoopCount < bufferLength; vectorLoopCount++)
+                {
+                    compute_generic_nn_interpolation_pkd3_to_pkd3(srcY, srcX, &roiLTRB, dstPtrTemp, srcPtrChannel, srcDescPtr);
+                    compute_warp_perspective_src_loc_next_term(vectorLoopCount, locW, locY, locX, srcY, srcX, perspectiveMatrix_f9, roiHalfHeight, roiHalfWidth);
+                    dstPtrTemp += 3;
+                }
+                dstPtrRow += dstDescPtr->strides.hStride;
+            }
+        }
+        else if ((srcDescPtr->layout == RpptLayout::NCHW) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp8u *dstPtrRow;
+            dstPtrRow = dstPtrChannel;
+
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp8u *dstPtrTemp;
+                dstPtrTemp = dstPtrRow;
+
+                int vectorLoopCount = 0;
+                Rpp32f locX, locY, locW, srcX, srcY;
+                __m256 plocX, plocY, plocW, pSrcX, pSrcY;
+                compute_warp_perspective_src_loc_params(i, vectorLoopCount, locW, locY, locX, perspectiveMatrix_f9, roiHalfHeight, roiHalfWidth);
+                plocX = _mm256_add_ps(_mm256_set1_ps(locX), pPerspectiveMatrixTerm0);
+                plocY = _mm256_add_ps(_mm256_set1_ps(locY), pPerspectiveMatrixTerm3);
+                plocW = _mm256_add_ps(_mm256_set1_ps(locW), pPerspectiveMatrixTerm6);
+                pSrcY = _mm256_add_ps(_mm256_div_ps(plocY, plocW), roiHalfHeightVec);
+                pSrcX = _mm256_add_ps(_mm256_div_ps(plocX, plocW), roiHalfWidthVec);
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    Rpp8u *dstPtrTempChn, *srcPtrTempChn;
+                    srcPtrTempChn = srcPtrChannel;
+                    dstPtrTempChn = dstPtrTemp;
+                    compute_generic_nn_srclocs_and_validate_avx(pSrcY, pSrcX, pRoiLTRB, pSrcStrideH, srcLoc, invalidLoad);
+                    for(int c = 0; c < srcDescPtr->c; c++)
+                    {
+                        __m256i pRow;
+                        rpp_simd_load(rpp_generic_nn_load_u8pln1_avx, srcPtrTempChn, srcLoc, invalidLoad, pRow);
+                        rpp_storeu_si64((__m128i *)(dstPtrTempChn), _mm256_castsi256_si128(pRow));
```

Comment:
Please see if static_cast can work here
```suggestion
                        rpp_storeu_si64(static_cast<__m128i *>(dstPtrTempChn), _mm256_castsi256_si128(pRow));
```

Reply:
Updated with reinterpret cast